File: frontend/src/models/solicitorSummary.ts
export default interface SolicitorSummary {
    name : string,
    shortDescription : string | null,
    id : string,
    rating : Rating | null
};

export interface Rating {
    value : number,
    maximum : number,
    provider : string
}

export interface Location {
    address : string,
    phone : string,
    ratings : Rating[]
}

export interface SolicitorInfo {
    name : string,
    shortDescription : string | null,
    id : string,
    phone? : string,
    email? : string,
    website? : string,

    ratings : Rating[],
    locations : Location[]
}

File: backend/libs/Solicitors.Core/Models/View/SolicitorSummary.cs
namespace Solicitors.Core.Models.View;

public record SolicitorSummary
{
    internal SolicitorSummary(Solicitor solicitor, string ratingsProvider)
    {
        Id = solicitor.SolicitorId;
        Name = solicitor.Name;
        ShortDescription = solicitor.ShortDescription;
        var rating = solicitor.Ratings.FirstOrDefault(x => x.Provider == ratingsProvider);
        if (rating is not null)
            Rating = new RatingInfo(rating);
    }

    public Guid Id { get; }
    public string Name { get; }
    public string? ShortDescription { get; }
    public RatingInfo? Rating { get; }
}